torch/quantization/qconfig.py

```diff
@@ -138,3 +138,36 @@ def assert_valid_qconfig(qconfig: Optional[Union[QConfig, QConfigDynamic]],
         )
         assert not is_per_channel, \
             'Per channel weight observer is not supported yet for ConvTranspose{n}d.'
+
+QConfigAny = Union[QConfig, QConfigDynamic, None]
+
+# should probably be in qconfig_utils but get circular reference issues TODO
+def add_device_to_obs_ctr_in_qconfig(qconfig: QConfigAny, module: torch.nn.Module):
```

Comment:
This unfortunately does couple device with `qconfig`, because we are encoding `QConfig` with the device and saving it in a object of type `QConfig`.  Could we estimate the engineering time to keep device out of qconfig and specify it directly when objects are constructed, so it's consistent with how other things work in PyTorch?

If we do really need this approach, could we move this logic out of anything related to `QConfig`, and name the resulting object differently (`QConfigAndDevice`, etc) to emphasize that this is no longer just quantization configs, there is an additional parameter here unrelated to quantization

Reply:
how about the new version that only ties it to module? is that philosophically at least an acceptable direction? Implementation aside?